File: java/Array&Hashing/LongestConsecutiveSequence.java
import java.util.HashSet;
import java.util.Set;

public class LongestConsecutiveSequence {
    /*
     * 128. Longest Consecutive Sequence
     * Difficulty: Medium
     * Topics: Array, Hash Table, Union Find
     */
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int best = 0;
        for (int num : nums) {
            set.add(num);
        }

        for (int n : set) {
            if (!set.contains(n-1)) { // only check or one direction
                int next = n + 1;
                while (set.contains(next)) {
                    next++;
                }
                best = Math.max(best, next - n);
            }

        }
        return best;
    }
}

File: python/Array&Hashing/LongestConsecutiveSequence.py
"""
128. Longest Consecutive Sequence
Difficulty: Hard
Topics: Array, Hash Table, Union Find
"""
def longestConsecutive(nums: list[int]) -> int:
    nums = set(nums)
    longest = 0
    for num in nums:
        if num - 1 not in nums: 
            # num is the start of a sequence because num-1 does not exist
            next_num = num + 1 # next_num is the next number in a sequence
            while next_num in nums: # if the sequence has a next number
                next_num += 1
            longest = max(longest, next_num - num)
    return longest